cmd/kubeadm/app/phases/upgrade/migrate.go

```diff
@@ -0,0 +1,135 @@
+/*
```

Comment:
i think we should name this file `preflight.go` if are going to keep it under upgrade.
the alternative location is `cmd/kubeadm/app/preflight`, but i'm leaning towards leaving it in this package.

possibly needs discussion.